### PR TITLE
Bug 1850707: ovirt: backports to 4.5

### PR DESCRIPTION
**Pull request by dougsland**
Bug 1850707: ovirt: IPI Improvements
Bug 1850723: ovirt: use pointer instead of struct for user/pass

### CODE REVIEW

**Comment by sdodson** [comment]
Needs lgtm and approved.

File: pkg/asset/installconfig/ovirt/credentials.go [hunk 1]
```diff
@@ -1,92 +1,231 @@
...
-	err = survey.Ask([]*survey.Question{
-		{
-			Prompt: &survey.Password{
-				Message: "oVirt engine password",
-				Help:    "",
-			},
-			Validate: survey.ComposeValidators(survey.Required, authenticated(&c)),
-		},
-	}, &c.Password)
+	// By default, we set Insecure true
```
Comment (rgolangh) [inline pkg/asset/installconfig/ovirt/credentials.go:180]: -1
why is that changed?

File: pkg/asset/installconfig/ovirt/credentials.go [hunk 2]
```diff
@@ -1,92 +1,231 @@
...
+	}
+	defer os.Remove(tmpFile.Name())
+
+	// Download PEM
+	httpResource.saveFilePath = tmpFile.Name()
+	httpResource.skipVerify = true
+	httpResource.urlAddr = engineConfig.PemURL
+	err = httpResource.downloadFile()
+	if err != nil {
+		logrus.Warning("cannot download PEM file from Engine!", err)
```
Comment (rgolangh) [inline pkg/asset/installconfig/ovirt/credentials.go:214]: So a failure to download the pem means we just ignore it from now on?

**Comment by sdodson** [comment]
/hold
What's going on with the second commit? It references a 4.6 bug that's unfixed. Is that fixed in master branch? If so where?

File: pkg/asset/installconfig/ovirt/credentials.go (hunk 1 above)
Comment (dougsland) [reply pkg/asset/installconfig/ovirt/credentials.go:180]: Because we cannot assume customer have their ssl working by default. We must check and detect first.

File: pkg/asset/installconfig/ovirt/credentials.go (hunk 2 above)
Comment (dougsland) [reply pkg/asset/installconfig/ovirt/credentials.go:214]: If failure customer will be able to continue or not (with a big warning in their screen.) bye CTRL C

Comment (rgolangh) [reply pkg/asset/installconfig/ovirt/credentials.go:214]: This should be a question if to proceed. Ctrl C means they don't know what's the state of their RHV/oVirt cluster after that. They have resources created or not. That's not a desired experience.

Comment (dougsland) [reply pkg/asset/installconfig/ovirt/credentials.go:214]: > This should be a question if to proceed. Ctrl C means they don't know what's the state of their RHV/oVirt cluster after
> that. They have resources created or not. That's not a desired experience.

fell free to open a card on that... we are talking here 4.5 branch not master. The patch was there in master for review and get approved.

**Comment by dougsland** [comment]
> /hold
> What's going on with the second commit? It references a 4.6 bug that's unfixed. Is that fixed in master branch? If so where?

as we talked over slack, already fixed.

File: pkg/asset/installconfig/ovirt/credentials.go (hunk 2 above)
Comment (sdodson) [reply pkg/asset/installconfig/ovirt/credentials.go:214]: > fell free to open a card on that... we are talking here 4.5 branch not master. The patch was there in master for review and get approved.

in that regard it's helpful if the PR links to the upstream PR so that it's clear whether or not this was previously discussed. If the bot were used to backport via `/cherrypick release-4.5` the bot would've linked to the upstream PR.

**Comment by sdodson** [comment]
> /test e2e-aws-scaleup-rhel7

There's no need to spam this test it's not blocking.

**Comment by rgolangh** [comment]
/hold 
This commit can't make it as is per the changes requested above, and also the 2nd commit bug is still in POST and we will have a problem verifying this in time.

**Comment by sdodson** [comment]
Per https://github.com/openshift/installer/pull/3801#issuecomment-651600216
/lgtm cancel

**Comment by dougsland** [comment]
Should be covered via KCS in 4.5
